packages/react-components/lib/notifications-dialog.tsx

```diff
@@ -0,0 +1,247 @@
+    display: 'grid',
+    gridTemplateColumns: 'repeat(2, 1fr) 4fr 1fr',
+    textAlign: 'center',
+    padding: theme.spacing(1),
+    width: '100%',
+    margin: `0.5rem 0`,
+  },
+  filter: {
+    display: 'flex',
+    float: 'right',
+    marginBottom: theme.spacing(1),
+  },
+  select: {
+    width: '150px',
+    padding: '0 0.5rem',
+  },
+  removeNotificationIcon: {
+    color: theme.palette.error.main,
+    padding: '0',
+  },
+  legend: {
+    display: 'flex',
+  },
+  indicator: {
+    padding: '0',
+    fontWeight: 600,
+  },
+  placeholder: {
+    color: theme.palette.success.main,
+  },
+}));
+
+const severityStyles = makeStyles((theme) => ({
+  high: {
+    color: theme.palette.secondary.dark,
+  },
+  medium: {
+    color: theme.palette.error.main,
+  },
+  low: {
+    color: theme.palette.warning.light,
+  },
+}));
+
+const SeverityIndicator = (props: SeverityIndicatoryProps): JSX.Element => {
+  const { severity, className } = props;
+  const classes = severityStyles();
+
+  const getStatusLabelClass = (severity: string): string => {
+    switch (severity) {
+      case 'High':
+        return classes.high;
+      case 'Medium':
+        return classes.medium;
+      case 'Low':
+        return classes.low;
+      default:
+        return '';
+    }
+  };
+
+  const styles = `${getStatusLabelClass(severity)} ${className}`;
+  return (
+    <Typography variant="body1" className={styles} align="left">
+      {severity}
+    </Typography>
+  );
+};
+
+export const NotificationsDialog = (props: NotificationDialogProps): JSX.Element => {
+  const classes = useStyles();
+
+  const { showNotificationsDialog, setShowNotifications, notifications } = props;
+
+  const [level, setLevel] = React.useState('');
+  const [rmfNotifications, setRmfNotifications] = React.useState(notifications);
+  // a copy of rmfNotifications
+  // ensure notifications can be restored to correct state after filtering
+  const [notficationsCopy, setNotificationsCopy] = React.useState(notifications);
+  // list of alert level for filtering
+  const alertLevel = React.useMemo(() => {
+    const holder: string[] = [];
+    notifications.forEach((notification) => {
+      if (!holder.includes(notification.severity)) {
+        holder.push(notification.severity);
+      }
+    });
+    return [...holder, 'All'];
+  }, [notifications]);
+
+  // handle filter change
+  const handleChange = (e: React.ChangeEvent<SelectChangeEvent>) => {
+    const val = e.target.value as string;
+    setLevel(val);
+    const filterNotifications: Notification[] = [];
+
+    if (val === 'All') {
+      setRmfNotifications(notficationsCopy);
+    } else {
+      notficationsCopy.forEach((notification) => {
+        if (notification.severity === val) {
+          filterNotifications.push(notification);
+        }
+      });
+      setRmfNotifications(filterNotifications);
+    }
+  };
+
+  // delete notifications once marked read
+  const deleteReadNotifications = (i: number) => {
```

Comment:
I suggest making this a prop, I think this implementation wouldn't work when the parent component reopens the dialog, the deleted notifications will come back.

Reply:
Will there be a back end sending deleted notifications or will a reducer implementation suffice for now?

Reply:
Yes, we will need to keep track of all notifications status in a database. This should not affect this component though, the job of updating the state should be left to the parent component, it will do whatever is needed and send a new `notifications` props without the deleted notifications.